frontend/src/components/ViewActivities/editActivity.js

```diff
@@ -58,11 +66,20 @@ class EditActivity extends React.Component {
 
     newVals[DB.ACTIVITIES_START_COUNTRY] = 
       getRefValue(this.editStartLocRef, 'No Change', activity[DB.ACTIVITIES_START_COUNTRY]);
-    newVals[DB.ACTIVITIES_END_COUNTRY] = 
-      getRefValue(this.editEndLocRef, 'No Change', activity[DB.ACTIVITIES_END_COUNTRY]);
+    
+    // Ref doesn't keep track of checked value?
+    newVals[DB.ACTIVITIES_FLIGHT] = this.state.flightCheck;
     
     newVals[DB.ACTIVITIES_START_TZ] = getRefValue(this.editStartTzRef);
-    newVals[DB.ACTIVITIES_END_TZ] = getRefValue(this.editEndTzRef);
+
+    if (newVals[DB.ACTIVITIES_FLIGHT]) { // This is a flight.
```

Comment:
This comment and the one on the else are unnecessary in my opinion.

Reply:
I do prefer them, makes things clearer.